include/deal.II/sundials/arkode.h

```diff
@@ -0,0 +1,858 @@
+                      const VectorType &fcur,
+                      const VectorType &rhs,
+                      VectorType &dst)> solve_jacobian_system;
+
+
+    /**
+     * Setup Mass matrix. This function is called by ARKode any time a mass
+     * matrix update is required. The user should compute the mass matrix (or
+     * update all the variables that allow the application of the mass matrix).
+     * This function is called by ARKode once, before any call to
+     * solve_mass_system().
+     *
+     * ARKode supports the case where the mass matrix may depend on time, but
+     * not the case where the mass matrix depends on the solution itself.
+     *
+     * If the user does not provide a solve_mass_matrix() function, then the
+     * identity is used. If the setup_mass() function is not provided, then
+     * solve_mass_system() should do all the work by itself.
+     *
+     * If the user uses a matrix based computation of the mass matrix, than
+     * this is the right place where an assembly routine shoulde be called to
+     * assemble both a matrix and a preconditioner. Subsequent calls (possibly
+     * more than one) to solve_mass_system() can assume that this function
+     * has been called at least once.
+     *
+     * Notice that no assumption is made by this interface on what the user
+     * should do in this function. ARKode only assumes that after a call to
+     * setup_mass() it is possible to call solve_mass_system(), to
+     * obtain a solution $x$ to the system $M x = b$.
+     *
+     * This function should return:
+     * - 0: Success
+     * - >0: Recoverable error (ARKodeReinit will be called if this happens, and
+     *       then last function will be attempted again
+     * - <0: Unrecoverable error the computation will be aborted and an assertion
+     *       will be thrown.
+     */
+    std::function<int(const double t)> setup_mass;
+
+    /**
+     * Solve the mass matrix linear system. This function will be called by
+     * ARKode (possibly several times) after setup_mass() has been called at
+     * least once. ARKode tries to do its best to call setup_mass() the minimum
+     * amount of times.
+     *
+     * A call to this function should store in `dst` the result of $M^{-1}$
+     * applied to `src`, i.e., `M*dst = src`. It is the users responsability to
+     * set up proper solvers and preconditioners inside this function.
+     *
+     * This function should return:
+     * - 0: Success
+     * - >0: Recoverable error (ARKodeReinit will be called if this happens, and
+     *       then last function will be attempted again
+     * - <0: Unrecoverable error the computation will be aborted and an assertion
+     *       will be thrown.
+     */
+    std::function<int(const VectorType &rhs, VectorType &dst)> solve_mass_system;
+
+    /**
+     * Process solution. This function is called by ARKode at fixed time steps,
+     * every `output_period` seconds, and it is passed a polynomial
```

Comment:
is output period in time units or time steps? If it's the former, then the time increment has to be consistent with the output period?

Reply:
The explanation is below the phrase you are pointing to... :)